File: app/src/main/java/eu/toma/dev/playground/warcraft/mounts/Mount.java
package eu.toma.dev.playground.warcraft.mounts;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Mount {

    @SerializedName("name")
    @Expose
    public String name;
    @SerializedName("spellId")
    @Expose
    public Integer spellId;
    @SerializedName("creatureId")
    @Expose
    public Integer creatureId;
    @SerializedName("itemId")
    @Expose
    public Integer itemId;
    @SerializedName("qualityId")
    @Expose
    public Integer qualityId;
    @SerializedName("icon")
    @Expose
    public String icon;
    @SerializedName("isGround")
    @Expose
    public Boolean isGround;
    @SerializedName("isFlying")
    @Expose
    public Boolean isFlying;
    @SerializedName("isAquatic")
    @Expose
    public Boolean isAquatic;
    @SerializedName("isJumping")
    @Expose
    public Boolean isJumping;
}

File: app/src/main/kotlin/warcraft/mounts/Mount.kt
package warcraft.mounts

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class Mount {
    @SerializedName("name")
    @Expose
    lateinit var name: String

    @SerializedName("spellId")
    @Expose
    var spellId: Int? = null

    @SerializedName("creatureId")
    @Expose
    var creatureId: Int? = null

    @SerializedName("itemId")
    @Expose
    var itemId: Int? = null

    @SerializedName("qualityId")
    @Expose
    var qualityId: Int? = null

    @SerializedName("icon")
    @Expose
    lateinit var icon: String

    @SerializedName("isGround")
    @Expose
    var isGround: Boolean? = null

    @SerializedName("isFlying")
    @Expose
    var isFlying: Boolean? = null

    @SerializedName("isAquatic")
    @Expose
    var isAquatic: Boolean? = null

    @SerializedName("isJumping")
    @Expose
    var isJumping: Boolean? = null
}